app/models/foreman_azure_rm/azure_rm_compute.rb

```diff
@@ -0,0 +1,240 @@
+      @azure_vm.resource_group = setrg
+    end
+
+    def vm_size
+    end
+
+    def vm_size=(setvsize)
+      @azure_vm.hardware_profile.vm_size = setvsize
+    end
+
+    def platform
+    end
+
+    def platform=(setplfrm)
+      @azure_vm.platform = setplfrm
+    end
+
+    def username
+    end
+
+    def username=(setuname)
+      @azure_vm.username = setuname
+    end
+
+    def password
+    end
+
+    def password=(setpswd)
+      @azure_vm.password = setpswd
+    end
+
+    def ssh_key_data
+    end
+
+    def os_disk_size
+    end
+
+    def os_disk_size=(setosdisk)
+      @azure_vm.os_disk.disk_size_gb = setosdisk
+    end
+
+    def premium_os_disk
+    end
+
+    def os_disk_caching
+    end
+
+    def os_disk_caching=(setcaching)
+      @azure_vm.os_disk.caching = setcaching
+    end
+
+    def puppet_master
+    end
+
+    def script_command
+    end
+
+    def script_uris
+    end
+
+    def custom_data
+    end
+
+    def image_id
+    end
+
+    def bridge
+    end
+
+    def instance_view
+      @azure_vm.instance_view
+    end
+
+    def wait_for(_timeout = 0, _interval = 0, &block)
+      instance_eval(&block)
+      return true
+    end
+
+    def ready?
+      vm_status == 'running'
+    end
+
+    def state
+      vm_status
+    end
+
+    def vm_status
+      sdk.check_vm_status(resource_group, name)
+    end
+
+    def interfaces_attributes=(attrs)
+      ;
+    end
+
+    def network_interface_card_ids
+      nics = @azure_vm.network_profile.network_interfaces
+      nic_card_ids = []
+      nics.each do |nic|
+        nic_card_ids << nic.id
+      end
+      nic_card_ids
+    end
+
+    def provisioning_ip_address
+      interfaces.each do |nic|
+        nic.ip_configurations.each do |configuration|
+          next unless configuration.primary
+          if configuration.public_ipaddress.present?
+            ip_id     = configuration.public_ipaddress.id
+            ip_rg     = ip_id.split('/')[4]
+            ip_name   = ip_id.split('/')[-1]
+            public_ip = sdk.public_ip(ip_rg, ip_name)
+            return public_ip.ip_address
+          else
+            return configuration.private_ipaddress
+          end
+        end
+      end    
+    end
+
+    def interfaces
+      interfaces = []
+      unless network_interface_card_ids.nil?
+        network_interface_card_ids.each do |nic_id|
+          nic_rg   = nic_id.split('/')[4]
+          nic_name = nic_id.split('/')[-1]
+          interfaces << sdk.vm_nic(nic_rg, nic_name)
+        end
+      end
+      interfaces
+    end
+
+    # get_public_ip and get_nic methods placed in adapter file.
+
+    def interfaces=(setifaces)
+      @azure_vm.network_profile.network_interfaces = setifaces
+    end
+
+    def ip_addresses
+      []
+    end
+
+    def destroy
+      return nil
+    end
+
+    def identity
+      @azure_vm.name
+    end
+
+    def identity=(setuuid)
+      @azure_vm.identity = setuuid
+    end
+
+    def self.parse(vm)
```

Comment:
I don't think we use this method anywhere.

Reply:
Actually we are using it here in [azure_rm.rb](https://github.com/theforeman/foreman_azure_rm/pull/36/files#diff-68325fa2c0d312fe64c53a3119247253R441)

Reply:
Commented in `azure_rm.rb`